robyn/openapi.py

```diff
@@ -0,0 +1,275 @@
+        according to spec
+        """
+        # robyn has paths like /:url/:etc whereas openapi requires path like /{url}/{path}
+        # this function is used for converting path params to the required form
+        # initialized with endpoint for handling endpoints without path params
+        endpoint_with_path_params_wrapped_in_braces = endpoint
+
+        endpoint_path_params_split = endpoint.split(":")
+        openapi_parameter_object = []
+
+        if len(endpoint_path_params_split) > 1:
+            endpoint_without_path_params = endpoint_path_params_split[0]
+
+            endpoint_with_path_params_wrapped_in_braces = (
+                endpoint_without_path_params[:-1] if endpoint_without_path_params.endswith("/") else endpoint_without_path_params
+            )
+
+            for path_param in endpoint_path_params_split[1:]:
+                path_param_name = path_param[:-1] if path_param.endswith("/") else path_param
+
+                openapi_parameter_object.append(
+                    {
+                        "name": path_param_name,
+                        "in": "path",
+                        "required": True,
+                        "schema": {"type": "string"},
+                    }
+                )
+                endpoint_with_path_params_wrapped_in_braces += "/{" + path_param_name + "}"
+
+        if query_params:
+            for query_param in query_params.__annotations__:
+                # ugly hack!
+                # returns "<class 'int'>" -- this line strips out the type a.k.a int from it
+                query_param_type = re.findall("'[a-z]+'", str(query_params.__annotations__[query_param]))[0].replace("'", "")
+
+                openapi_parameter_object.append(
+                    {
+                        "name": query_param,
+                        "in": "query",
+                        "required": False,
+                        "schema": {"type": query_param_type},
+                    }
+                )
+
+        return endpoint_with_path_params_wrapped_in_braces, {
+            "summary": summary,
+            "tags": tags,
+            "parameters": openapi_parameter_object,
+            "responses": {
+                "200": {
+                    "description": "Successful Response",
+                }
+            },
+        }
+
+    def dump_openapi_spec_file(self) -> str:
+        """
+        Write the current openapi spec dictionary to openapi.json file
+        @return: a JSON string representing the openapi spec
+        """
+        return json.dumps(self.openapi_spec)
+
+    def docs_handler(self) -> FileResponse:
+        """
+        Handler to the swagger html page to be deployed to the endpoint `/docs`
+        side effect: this function also dumps the openapi json file
+        @return: the swagger html page
+        """
+        self.dump_openapi_spec_file()
```

Comment:
this doesn't do anything, just returns a string. Where are you dumping the file?

Reply:
🔴

Reply:
I have updated the implementation. I was assuming something else when I fisr wrote the code, my mistake!